types/ui-grid/index.d.ts

```diff
@@ -3799,7 +3799,7 @@ declare namespace uiGrid {
          * If suppressRemoveSort is also set, the unsorted state will be skipped even if it is listed here. Each direction may
          * not appear in the list more than once (e.g. [ASC, DESC, DESC] is not allowed), and the list may not be empty.*
          */
-        sortDirectionCycle?: Array<IUiGridConstants>;
+        sortDirectionCycle?: Array<null | string>;
```

Comment:
This fixes the problem, but could be even better by making the type even less general than `string`, something like `[null, uiGridConstants.ASC, uiGridConstants.DESC]`.